lib/pbench/server/database/models/datasets.py

```diff
@@ -730,8 +885,165 @@ def create(**kwargs) -> "Metadata":
         else:
             return meta
 
+    @staticmethod
+    def is_key_path(key: str, valid: List[str]) -> bool:
+        """
+        Determine whether 'key' is a valid Metadata key path using the list
+        specified in 'valid'. If the specified key is in the list, then it's
+        valid. If the key is a dotted path and the first element plus a
+        trailing ".*" is in the list, then this is an open key namespace where
+        any subsequent path is acceptable: e.g., "user.*" allows "user", or
+        "user.contact", "user.contact.name", etc.
+
+        Args:
+            key: metadata key path
+            valid: list of acceptable key paths
+
+        Returns:
+            True if the path is valid, or False
+        """
+        k = key.lower()
+        # Check for exact match
+        if k in valid:
+            return True
+        path = k.split(".")
+        # Disallow ".." and trailing "."
+        if "" in path:
+            return False
+        # Check for open namespace match
+        if not path[0] + ".*" in valid:
```

Comment:
If you change this PR for some other reason, please consider changing this line to
```
        if path[0] + ".*" not in valid:
```
which is clearer, more Pythonic, and [PEP-8-recommended](https://www.python.org/dev/peps/pep-0008/#id49).